### PR TITLE
Fix segfault in PKCS7 test

### PR DESCRIPTION
**Pull request by justsmth**
### Description of changes: 
Fix Segfault in PKCS7 test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

**Review by WillChilds-Klein — DISMISSED** [review]

File: crypto/pkcs7/pkcs7_test.cc [hunk 1]
```diff
@@ -1830,15 +1828,15 @@ TEST(PKCS7Test, TestEnveloped) {
   // expectation. Ideally we'd find a way to access the padded plaintext and
   // account for this deterministically by checking the random "padding" and
   // adusting accordingly.
-  int max_decrypt =
-      sizeof(decrypted) + EVP_CIPHER_block_size(EVP_aes_128_cbc());
-  int decrypted_len = BIO_read(bio.get(), decrypted, max_decrypt);
-  if (decrypted_len > (int)pt_len) {
+  const size_t max_decrypt =
+    pt_len + EVP_CIPHER_block_size(EVP_aes_128_cbc());
+  const size_t decrypted_len = (size_t)BIO_read(bio.get(), decrypted, sizeof(decrypted));
```
Comment (torben-hansen) [inline crypto/pkcs7/pkcs7_test.cc:1833]: Might be a little bit more complicated, because `BIO_read` can return negative failure codes.

Comment (justsmth) [reply crypto/pkcs7/pkcs7_test.cc:1833]: Ok, I'll add a check to ensure that the value is in the expected range:
```
ASSERT_LE(decrypted_len, sizeof(decrypted));
```

File: crypto/pkcs7/pkcs7_test.cc [hunk 2]
```diff
@@ -1814,7 +1812,7 @@ TEST(PKCS7Test, TestEnveloped) {
   int decrypt_ok =
       PKCS7_decrypt(p7.get(), rsa_pkey.get(), /*certs*/ nullptr, bio.get(),
                     /*flags*/ 0);
-  EXPECT_LE(sizeof(decrypted), BIO_pending(bio.get()));
+  EXPECT_LE(pt_len, BIO_pending(bio.get()));
   OPENSSL_cleanse(decrypted, sizeof(decrypted));
   // There's a fun edge case here for block ciphers using conventional PKCS#7
```
Comment (WillChilds-Klein) [inline crypto/pkcs7/pkcs7_test.cc:1817]:
```suggestion
  // There's a fun edge case here for block ciphers using conventional PKCS#1
```

**Review by WillChilds-Klein — APPROVED** [review]

File: crypto/pkcs7/pkcs7_test.cc (hunk 1 above)
Comment (torben-hansen) [reply crypto/pkcs7/pkcs7_test.cc:1833]: Ok with the note that `(size_t)BIO_read` converts signed to unsigned. So,
`ASSERT_LE(decrypted_len, sizeof(decrypted));` could be a bit awkward.

**Review by torben-hansen — APPROVED** [review]

**Pull request merged**